libmamba/tests/src/core/test_filesystem.cpp

```diff
@@ -349,6 +351,50 @@ namespace mamba
             fs::remove_all(tmp_dir);
             REQUIRE_FALSE(fs::exists(tmp_dir));
         }
+
+        TEST_CASE("create_cache_dir")
+        {
+            const auto cache_dir = fs::temp_directory_path() / "mamba-fs-cache-dir";
+            const auto set_gid_tmp_dir = fs::temp_directory_path() / "mamba-fs-set_gid_tmp_dir";
+
+            mamba::on_scope_exit _(
+                [&]
+                {
+                    fs::remove_all(cache_dir);
+                    fs::remove_all(set_gid_tmp_dir);
+                }
+            );
+
+            // Get the information about whether the filesystem supports the `set_gid` bit.
+            bool supports_setgid_bit = false;
+
+            std::error_code ec;
+            fs::permissions(set_gid_tmp_dir, fs::perms::set_gid, fs::perm_options::add, ec);
+
+            if (!ec)
+            {
+                supports_setgid_bit = (fs::status(set_gid_tmp_dir).permissions() & fs::perms::set_gid)
+                                      == fs::perms::set_gid;
+            }
+
+            create_cache_dir(cache_dir);
+
+            // Check that the permissions of `cache_dir` are _at least_ `rwxr-xr-x`
```

Comment:
It has been observed on Linux, that `rwx` is not necessarily set (for the group) even if explicitly specified (in which case `r-x` is set instead).